mmv1/products/compute/RouterNat.yaml

```diff
@@ -164,6 +184,7 @@ properties:
       imports: 'selfLink'
       description: 'A reference to an address associated with this NAT'
     custom_expand: 'templates/terraform/custom_expand/array_resourceref_with_validation.go.erb'
+    diff_suppress_func: 'hasInitialNatIpsDiffSuppress'
```

Comment:
Using the `default_from_api: true` parameter did seen to achieve the needed behavior, even with the current use of `send_empty_value: true`.
Only a small issue showed up during testing in that when changing from a `nat_ip_allocate_option = "MANUAL_ONLY"` config with the use of `nat_ips` to a config using `nat_ip_allocate_option = "AUTO_ONLY"`, the user may need to explicitly set `nat_ips = []`, else the provider tries to patch the router nat using the previous nat_ips values.

Reply:
> Only a small issue showed up during testing in that when changing from a nat_ip_allocate_option = "MANUAL_ONLY" config with the use of nat_ips to a config using nat_ip_allocate_option = "AUTO_ONLY", the user may need to explicitly set nat_ips = [], else the provider tries to patch the router nat using the previous nat_ips values.

This is a concern to be honest. I think it'd be worth adding an acceptance test that shows a google_compute_router_nat resource being created with `nat_ip_allocate_option = "MANUAL_ONLY"`  and use of google_compute_router_nat_address, and then being changed to `nat_ip_allocate_option = "AUTO_ONLY"` and setting `nat_ips = []`. If that acceptance test shows that the swap isn't possible due to problems setting `nat_ips = []` then I think we'll need to revert back using diff suppress funcs, instead of default_from_api: true.

If you set up the acceptance test I can run it on this PR to see how/if it fails and then we can figure out if reverting to using diff suppress funcs is the route forward.

Reply:
I made a change to `TestAccComputeRouterNat_update` so that the last two test steps shows this scenario, the `testAccComputeRouterNatUpdateToNatIPsName` config sets the router nat with `nat_ip_allocate_option = "MANUAL_ONLY"` and one value for `nat_ips`, while the `testAccComputeRouterNatBasicBeforeUpdate` config changes it for `nat_ip_allocate_option = "AUTO_ONLY"` and `nat_ips = []`.

Reply:
This issue only really pops up when using `nat_ips` from the router_nat directly, as when using a router_nat_address the router_nat is supposed to have only `initial_nat_ips` values configured.
The issue was showing up on the last two test steps of `TestAccComputeRouterNat_update` so I added `nat_ips = []` to the last step. Nevertheless I created the test TestAccComputeRouterNatAddress_withAutoAllocateAndAddressRemoved anyway just to be sure setting `nat_ips = []` does not cause any issues here.